src/components/Query.tsx

```diff
@@ -56,12 +56,15 @@ class QueryHandler extends Component<QueryHandlerProps, QueryHandlerState> {
     this.unsubscribe = teardown;
   };
 
-  state = {
-    executeQuery: this.executeQuery,
-    data: undefined,
-    error: undefined,
-    fetching: false,
-  };
+  constructor(props) {
+    super(props);
```

Comment:
I have further implemented the SSR approach but the sync would imply we execute our query here, this isn't possible since this.setState would result in a noop.